crypto/s2n_openssl_x509.c

```diff
@@ -114,5 +116,20 @@ S2N_RESULT s2n_openssl_x509_get_cert_info(X509 *cert, struct s2n_cert_info *info
     RESULT_GUARD_OSSL(OBJ_find_sigid_algs(info->signature_nid, &info->signature_digest_nid, NULL),
             S2N_ERR_CERT_TYPE_UNSUPPORTED);
 
+    DEFER_CLEANUP(EVP_PKEY *pubkey = X509_get_pubkey(cert), EVP_PKEY_free_pointer);
+    RESULT_ENSURE_REF(pubkey);
+
+    info->public_key_bits = EVP_PKEY_bits(pubkey);
+    RESULT_ENSURE(info->public_key_bits > 0, S2N_ERR_CERT_TYPE_UNSUPPORTED);
+
+    if (EVP_PKEY_base_id(pubkey) == EVP_PKEY_EC) {
+        const EC_KEY *ec_key = EVP_PKEY_get0_EC_KEY(pubkey);
+        RESULT_ENSURE_REF(ec_key);
+        info->public_key_nid = EC_GROUP_get_curve_name(EC_KEY_get0_group(ec_key));
```

Comment:
It might be good to check the return of `EC_KEY_get0_group` before giving it to the libcrypto here.